floris/tools/floris_interface.py

```diff
@@ -820,6 +820,42 @@ def get_farm_AEP(self, wd, ws, freq, yaw=None):
             AEP_sum = AEP_sum + self.get_farm_power()*freq[i]*8760
         return AEP_sum
 
+    def change_turbine(self, turb_num_array, turbine_change_dict):
+        """
+        Change turbine properties of given turbines
+
+        Args:
+            turb_num_array (list): list of turbines to apply the change to
+            turbine_change_dict (dict): dictionary of changes to make.  All 
+                key values should be from the JSON turbine/properties set.
+                Any key values not specified will be copied from the original
+                JSON values
+        """
+
+        #TODO This may not work as intended if floris_interface not initialized with JSON
+
+        # First re-read the input file to get the baseline parameters
+        json_dict = self.floris.input_reader._parseJSON(self.input_file)
+        turbine_dict = json_dict["turbine"]
+
+        # Now go through the turbine entries and change if they are in turbine change dict
+        for key in turbine_dict["properties"]:
+            if key in turbine_change_dict:
+                turbine_dict["properties"][key] = turbine_change_dict[key]
+                print("Changing key %s to the following value..." % key)
+                print('...,',turbine_dict["properties"][key])
+            else:
+                print('Leaving key %s as defined in input file %s' % (key,self.input_file))
+
+        # Now go through turbine list and re-init any in turb_num_array
+        for t_idx in turb_num_array:
+            print('Updating turbine: %00d' % t_idx)
+            self.floris.farm.turbines[t_idx].__init__(turbine_dict)
```

Comment:
Why call the `__init__` method like this?

Reply:
I'm not sure what actually happens when you set the turbines attribute on Farm since thats a virtual property with a getter-method only (`@propert`). I think you should use the turbine array in `TurbineMap` instead.

Reply:
If you try the example, it seems to fly, but I admit it seems precarious.  I started out using the TurbineMap, but it was trickier to say, set the second turbine, because TurbineMap is a dict keyed off coordinates, instead of indexed array I think?  Maybe I could work some more with you on doing this correctly?  This was just my first pass...

Reply:
also on the __init__ question, it's because the init function builds the interpolation function, if the user changes the cp/ct tables, these functions will need to be updated...

Reply:
I think we could either add an id tag to the turbines or actually keep an array in TurbineMap. That would be more explicit and avoid having two objects get out of sync.

Reply:
I'm not sure I follow, will you be at NREL next week and we can discuss live?